executor/wasm-host/src/host.rs

```diff
@@ -956,12 +965,10 @@ pub fn casper_env_caller<S: GlobalStateReader, E: Executor>(
 
     // TODO: Decide whether we want to return the full address and entity kind or just the 32 bytes
     // "unified".
-    let (entity_kind, data) = match &caller.context().caller {
-        Key::Account(account_hash) => (0u32, account_hash.value()),
-        Key::SmartContract(smart_contract_addr) => (1u32, *smart_contract_addr),
-        other => panic!("Unexpected caller: {other:?}"),
-    };
-    let mut data = &data[..];
+
+    let entity_addr = context_to_entity_addr(caller.context());
+    let entity_kind = if entity_addr.is_account() { 0u32 } else { 1u32 };
```

Comment:
```suggestion
    let entity_kind = if entity_addr.is_account() { EntityKindTag::Account } else { EntityKidnTag::Contract };
```

That should work, it's even in the same module